src/routes/data/handlers.ts

```diff
@@ -64,6 +65,70 @@ const setDataHeaders = ({
   res.header('Content-Length', data.size.toString());
 };
 
+const handlePartialDataResponse = (log: Logger, rangeHeader: string, res: Response, data: ContiguousData, dataAttributes: ContiguousDataAttributes | undefined ) => {
+  const totalSize = data.size;
+  const parts = rangeHeader.replace(/bytes=/, "").split("-");
+  const start = parseInt(parts[0], 10);
+  const end = parts[1] ? parseInt(parts[1], 10) : totalSize - 1;
+  const chunkSize = end - start + 1;
+
+  // Check if the range is valid
+  if (start >= 0 && end < totalSize && start <= end) {
```

Comment:
We should try to handle multiple ranges too since that's part of the spec. https://www.npmjs.com/package/byte-range-stream should work for that.

Reply:
I added the library but my testing with curl doesn't seem to be making the multi range work. I still just get back the content of the first part of the range